vscode-lean4/src/leanclient.ts

```diff
@@ -100,10 +101,13 @@ export class LeanClient implements Disposable {
     }
 
     async showRestartMessage(): Promise<void> {
-        const restartItem = 'Restart Lean Language Server';
-        const item = await window.showErrorMessage('Lean Language Server has stopped unexpectedly.', restartItem)
-        if (item === restartItem) {
-            void this.start();
+        if (!this.showingRestartMessage){
+            this.showingRestartMessage = true;
+            const restartItem = 'Restart Lean Language Server';
+            const item = await window.showErrorMessage('Lean Language Server has stopped unexpectedly.', restartItem)
+            if (item === restartItem) {
```

Comment:
Also need to add `this.showingRestartMessage = false;` here after window.showErrorMessage returns (remember it is async and blocks as long as the popup is visible).

Reply:
The `this.showingRestartMessage = false;` is inside start() (inside restart()) function (finishing the function), so we can be sure that the client was restarted correctly. Do we need this `this.showingRestartMessage = false; ` again after showing the error message?

Reply:
Fine, but it doesn't need to be that complicated, it can be right here on line 108.  This boolean is only to block multiple calls to this method from attempting to show the error message when it is already being shown.